src/lib/dateLib.ts

```diff
@@ -173,3 +178,206 @@ export const dateLib = {
+
+  /** Checks if the first date is after the second date. */
+  isAfter: DateLibBase["isAfter"] = (...args) =>
+    this.dateLibBase.isAfter(...args);
+
+  /** Checks if the first date is before the second date. */
+  isBefore: DateLibBase["isBefore"] = (...args) =>
+    this.dateLibBase.isBefore(...args);
+
+  /** Checks if the given value is a date. */
+  isDate: DateLibBase["isDate"] = (value): value is Date =>
+    this.dateLibBase.isDate(value);
+
+  /** Checks if the given dates are the same day. */
+  isSameDay: DateLibBase["isSameDay"] = (...args) =>
+    this.dateLibBase.isSameDay(...args);
+
+  /** Checks if the given dates are in the same month. */
+  isSameMonth: DateLibBase["isSameMonth"] = (...args) =>
+    this.dateLibBase.isSameMonth(...args);
+
+  /** Checks if the given dates are in the same year. */
+  isSameYear: DateLibBase["isSameYear"] = (...args) =>
+    this.dateLibBase.isSameYear(...args);
+
+  /** Returns the maximum of the given dates. */
+  max: DateLibBase["max"] = (...args) => this.dateLibBase.max(...args);
+
+  /** Returns the minimum of the given dates. */
+  min: DateLibBase["min"] = (...args) => this.dateLibBase.min(...args);
+
+  /** Sets the month for the given date. */
+  setMonth: DateLibBase["setMonth"] = (...args) =>
+    this.dateLibBase.setMonth(...args);
+
+  /** Sets the year for the given date. */
+  setYear: DateLibBase["setYear"] = (...args) =>
+    this.dateLibBase.setYear(...args);
+
+  /** Returns the start of the day for the given date. */
+  startOfDay: DateLibBase["startOfDay"] = (...args) =>
+    this.dateLibBase.startOfDay(...args);
+
+  /** Returns the start of an ISO week for the given date. */
+  startOfISOWeek: DateLibBase["startOfISOWeek"] = (...args) =>
+    this.dateLibBase.startOfISOWeek(...args);
+
+  /** Returns the start of the month for the given date. */
+  startOfMonth: DateLibBase["startOfMonth"] = (...args) =>
+    this.dateLibBase.startOfMonth(...args);
+
+  /** Returns the start of the year for the given date. */
+  startOfYear: DateLibBase["startOfYear"] = (...args) =>
+    this.dateLibBase.startOfYear(...args);
+
+  // ================================================
+  // DateLib functions which require locale overrides
+  // ================================================
+
+  /** Returns the end of the week for the given date. */
+  endOfWeek = <DateType extends Date>(date: DateArg<DateType>) => {
+    if (!this.locale) {
+      throw new Error(
+        "DateLib needs to be initialized with locale to use the `endOfWeek` function."
+      );
+    }
+
+    return endOfWeek(date, this.options);
+  };
+
+  /** Formats the given date using the specified format string. */
+  format = (date: DateArg<Date> & {}, formatStr: string) => {
+    if (!this.locale) {
+      throw new Error(
+        "DateLib needs to be initialized with locale to use the `format` function."
+      );
+    }
+
+    return format(date, formatStr, this.options);
+  };
```

Comment:
The assumption here is that `this.options` includes `this.locale`, right? If so, why is `this.locale` needed separately? Couldn't we just:

```suggestion
    if (!this.options.locale) {
      throw new Error(
        "DateLib needs to be initialized with locale to use the `format` function."
      );
    }

    return format(date, formatStr, this.options);
  };
```

P.S. I love how well the code is documented. Thanks!

Reply:
yeah absolutely! I think what you're suggesting is more clear, I was just using `this.locale` as shorthand, but it's more indirection.